enhancements/insights/pulling-pull-secret-from-ocm.md

```diff
@@ -0,0 +1,168 @@
+4. The Insights operator calls this API periodically, fetches the `pull-secret` and compares it with the current global `pull-secret` on the cluster.
+5. If there is a mismatch, the Insights operator applies the new `pull-secret`.
+   * Note: MCO updating the `pull-secret` on the nodes will not result in node reboots.
+6. The Insights operator itself starts using the new `pull-secret`
+7. Cluster comes up with the new `pull-secret` and starts reporting back to OCM into the new account.
+
+Note: This process will work only on the clusters where the feature was released as part of the Insights operator and will not be backported to older versions.
+
+### User Stories
+
+As an OCM user, I would like to transfer cluster ownership from my account to another user's account in one step.
+
+### API Extensions
+
+No API extensions.
+
+### Risks and Mitigations
+
+Risk: The OCM API is down or returning HTTP response code >= 500.
+
+Mitigation: The Insights Operator is marked as degraded in such case.
+
+Risk: The `pull-secret` does not work or is otherwise corrupted.
+
+Mitigation: We must refer to the OCP documentation, which describes the manual fix procedure. 
+
+
+## Design Details
+
+### Authorization
+
+The Insights Operator is able to pull the updated `pull-secret` content from the OCM API using the existing `cloud.openshift.com` token
+available in the `pull-secret` (in the `openshift-config` namespace).
+
+The Insights Operator must provide a cluster ID as an identifier of the cluster.
+
+### OCM API
+
+The OCM API provides an endpoint that will serve the complete `pull-secret` definition when the migration window is opened. 
+The migration window will be opened for certain period of time (e.g 5 days) so tha the Insights Operator can safely update the secret.
+
+Open question: What exactly will the API return outside of the given migration window?  
+
+### Monitoring
+
+There will be a new metric registered by the Insights Operator, which will count the number of responses (according the HTTP response status code) from the OCM API.
+
+
+### Test Plan
+
+- `insights-operator-e2e-tests` suite can verify that the Insights Operator is able to download and update the `pull-secret` content
+
+### Graduation Criteria
+
+Thera are no graduation criteria. This feature is planned as a fast track feature.
+
+#### Dev Preview -> Tech Preview
+
+Empty
+
+#### Tech Preview -> GA
+
+Empty
+
+#### Removing a deprecated feature
+
+Removing this feature will require updating the Insights operator code base.
+
+### Upgrade / Downgrade Strategy
+
+There is no upgrade/downgrade strategy needed.
+
+### Version Skew Strategy
+
+There is no Skew strategy needed. This work should have no impact on the upgrade. It doesn't require any coordinated behavior in the control plane. No other components will change.
```

Comment:
Assuming the in-cluster pull secret location doesn't move around, this seems safe.  But possibly worth talking through skew between different versions of the Insights operator and the OCM API service.  Or is the expectation that we get that API right on the first try and never have to update it going forward?

Reply:
Yes I think we should say that the OCM API should preserve the backport compatibility and that the new version of the API will likely require an update of the Insights operator code. Would it be sufficient?

Reply:
I agree the probability of the API changing during the lifetime of the OCP version being supported should be very small and we should demand on the OCM API preserve the compatibility through the life-time of the versions depending on that particular API.

I can imagine having additional logic on specific extra status code form the API (let's say 410 Gone) and a follow-up alert from the insights-operator that it's outdated and the user needs to update for it to work properly: but personally, given its probability and the other ways there are in place notifying the user on using out-dated cluster, I would vote for simple logic and just assume the API will work on supported versions.

Reply:
I updated this chapter. Please check.

Reply:
b55db2b012f9 looks good to me; thanks :)

Reply:
Thanks too.